modal-js/src/app.ts

```diff
@@ -82,4 +85,19 @@ export class App {
   async imageFromRegistry(tag: string): Promise<Image> {
     return await fromRegistryInternal(this.appId, tag);
   }
+
+  async imageFromAwsEcr(tag: string, secret: Secret): Promise<Image> {
+    if (!secret.secretId) {
+      throw new Error(
+        "secret must be a reference to an existing Secret, e.g. `await Secret.from_name('my_secret')`",
+      );
+    }
```

Comment:
I don't think this validation is needed or helpful since the `Secret` object, once constructed properly, must have a secret ID already.

Reply:
Ah got it. Yeah that would be caught by TypeScript! A more direct check might be

```ts
if (!(secret instanceof Secret)) {
  throw new TypeError("...");
}
```